### DISCUSSION TITLE
Support to run continuously logging data to a file

### DISCUSSION THREAD

**Discussion by ethernet802** [joshschmelzle/lswifi#18] [Ideas]
It would be good if we could have it run continuously scanning every x amount of seconds to gather data.

For example in my use case I would have 8 different types of antenna to test. I would like to gather data for each antenna based on a period of an hour or longer that would scan every 30 seconds.

I could then use that data to determine which antenna we should be using.

I am kind of doing this using powershell to run lswifi and then outputting the json to csv

thank you for developing this, I use it at my workplace to look for issues.

**Comment by joshschmelzle** [comment]
I've been thinking about ways to make lswifi do more than the one-shot scan it currently does.

For example, I'd like to be able to filter on a single particular BSSID and calculate and display the average, mean, standard deviation, and coefficient of variation of RSSI readings for a provided number of measurements.

I think under the hood, some of the changes required would also align with what you're asking for. What format are you looking for? And are you looking for all the results to be in a single file?

**Reply by ethernet802** [reply]
Yes that sounds good. 

I want to be able to do the usual filtering as you mention, I am not too fussed with the calculations as I want to be able to import the data into excel or a time series db and manipulate from there.

For my purposes a csv file would give me the most flexibility and the information currently outputting to cli would be perfect inlcuding the time of the scan.

DATETIME,SSID,BSSID,RSSI,PHY,CHANNEL,FREQ.,SS,SECURITY,AMENDMENTS,AP UPTIME   etc.

Each line could be appended to the csv file as the scan cycle completes.

A single file would be fine.

my ideal scenario would be to scan a particular bssid every 30 seconds or so and have the following line appended to a text file including date/time

date/time, ssid, bsssid, rssi, channel, freq , utilization

**Comment by joshschmelzle** [comment]
@ethernet802 I've made some changes in the `dev` branch if you want to try them out. 

New output options:

- `--csv` - output will be formatted as csv to a file in the current directory with a file named by lswifi
- `--csv <path>` - output will be formatted as csv to a file at the path provided
- `--json` - output will be formatted as json to a file in the current directory with a file named by lswifi
- `--json <path>` - output will be formatted as json to a file at the path provided

New timing options:

- `-i #, --interval #` - seconds between scans (so `--interval 30` if you want a scan to trigger roughly once every 30 seconds)
- `--time` - set test length in seconds to continually perform scans for (so `--time 3600` to keep lswifi up for 1 hour)
- `-n #, --scans #` - set how many scans to do before exiting

Steps:

- Clone repo to local machine
- Switch to dev branch e.g. `git checkout dev`
- Install with pip from root directory of the repo e.g. `py -m pip install .`

**Reply by ethernet802** [reply]
Thank you, we have another batch of antennas to test next week and will test it then and give feedback. Looks great!

**Reply by joshschmelzle** [reply]
lswifi 0.1.29 with support for this has been pushed to pypi.
